File: dart-language/charge-system-demo/client-ng-dart/lib/src/login/login_component.dart
import 'package:angular/angular.dart';
import 'package:angular_forms/angular_forms.dart';
import '../user_service.dart';

@Component(
  selector: 'my-login',
  styleUrls: ['login_component.css'],
  templateUrl: 'login_component.html',
  directives: [coreDirectives, formDirectives],
)
class LoginComponent {
  final UserService userService;
  String username = '';
  String password = '';

  LoginComponent(this.userService);

  void login() async {
    await this.userService.login(username, password);
    print(1);
  }

}

File: dart-language/charge-system-demo/client-ng-ts/projects/at-antd/src/app/login/login.component.ts
import { Component } from '@angular/core';
import { Router } from '@angular/router';
import { UserService } from '../user.service';
import { NzMessageService } from 'ng-zorro-antd';

@Component({
  selector: 'ant-login',
  templateUrl: './login.component.html',
  styleUrls: ['./login.component.less']
})
export class LoginComponent {
  username: string = '';
  password: string = '';

  constructor(
    private router: Router,
    private message: NzMessageService,
    private userService: UserService
  ) { }

  toDashboard(): void {
    this.router.navigateByUrl('/dashboard');
  }
  handleLogin() {
    const { username, password } = this;
    if (username === '' || password === '') {
      return this.message.create('error', 'username or password can not be empty');
    }
    this.userService.login(username, password).subscribe((res) => {
      if (res) { this.toDashboard(); }
    });
  }

}
